test/unit/vl-map-layer.test.html

```diff
@@ -100,31 +116,57 @@
+        await awaitUntil(() => {
+          let view = kaartElement.map.getView();
+          return isEqual(view.getCenter(), [147055, 197908]) && view.getResolution() === 128;
+        });
+      });
+
+      test('het aanroepen van zoomToExtent zal de kaart doen inzoomen op de bounding box', async () => {
+        let kaartElement = fixture('vl-map-fixture');
+        let kaartlaagElement = kaartElement.querySelector('vl-map-layer');
+        let view = kaartElement.map.getView();
+        assert.notDeepEqual(view.getCenter(), [147055, 197908]);
+
+        await kaartlaagElement.zoomToExtent();
+        assert.deepEqual(view.getCenter(), [147055, 197908]);
+        assert.equal(view.getResolution(), 8);
+
+        await kaartlaagElement.zoomToExtent(3);
+        assert.deepEqual(view.getCenter(), [147055, 197908]);
+        assert.equal(view.getResolution(), 128);
       });
       
-      test('bij het toevoegen van een feature zal deze automatisch toegevoegd worden aan de laag en zal de kaart opnieuw zoomen naar de extent van de kaartlaag indien de auto-extent geactiveerd werd', () => {
+      test('bij het toevoegen van een feature zal deze automatisch toegevoegd worden aan de laag en zal de kaart opnieuw zoomen naar de extent van de kaartlaag indien de auto-extent geactiveerd werd', async () => {
         const kaartElement = fixture('vl-map-fixture');
         const kaartlaagElement = kaartElement.querySelector('vl-map-layer');
-        
+
         let layer = kaartElement.map.getOverlayLayers()[0];
         assert.lengthOf(layer.getSource().getFeatures(), 1);
         assert.equal(geoJSON.writeFeatures(layer.getSource().getFeatures()), kaartlaagElement.getAttribute('features'));
         assert.deepEqual(kaartElement.map.getView().getCenter(), [140860.69299028325, 190532.7165957574]);
-        
+
         kaartlaagElement.setAttribute('features', "{\"type\":\"FeatureCollection\",\"features\":[{\"type\":\"Feature\",\"geometry\":{\"type\":\"Point\",\"coordinates\":[147055,197908]},\"properties\":null},{\"type\":\"Feature\",\"geometry\":{\"type\":\"Point\",\"coordinates\":[157055,207908]},\"properties\":null}]}");
         kaartlaagElement.setAttribute('auto-extent', '');
-        layer = kaartElement.map.getOverlayLayers()[0];
-        assert.lengthOf(layer.getSource().getFeatures(), 2);
-        assert.equal(geoJSON.writeFeatures(layer.getSource().getFeatures()), kaartlaagElement.getAttribute('features'));
-        assert.notDeepEqual(kaartElement.map.getView().getCenter(), [140860.69299028325, 190532.7165957574]);
+        await awaitUntil(() => {
+          layer = kaartElement.map.getOverlayLayers()[0];
+          return layer.getSource().getFeatures().length === 2 &&
+                  geoJSON.writeFeatures(layer.getSource().getFeatures()) === kaartlaagElement.getAttribute('features') &&
+                  !isEqual(kaartElement.map.getView().getCenter(), [140860.69299028325, 190532.7165957574]);
```

Comment:
idem als hierboven, ik vind het een beetje raar om te awaiten op iets dat niet waar is ... hier snap ik het beter dan hierboven omdat het hier van waar naar niet-waar gaat ...

Reply:
Yes, dat kan. Ik had for now gewoon een awaitUntil geschreven rond de bestaande conditie. Zal dat aanpassen.